model/project.go

```diff
@@ -2296,3 +2297,55 @@ func GetVariantsAndTasksFromPatchProject(ctx context.Context, settings *evergree
 	}
 	return &variantsAndTasksFromProject, nil
 }
+
+// ReadOutputPath reads the content of a file and parses it into a github.CheckRunOutput struct
+// if it fails validation, it returns an error
+func ReadAndValidateOutputPath(file string) (*github.CheckRunOutput, error) {
+	checkRunOutput := &github.CheckRunOutput{}
+
+	if file == "" {
+		return checkRunOutput, nil
+	}
+
+	if err := utility.ReadJSONFile(file, &checkRunOutput); err != nil {
+		return nil, err
+	}
+
+	return checkRunOutput, validateCheckRuns(checkRunOutput)
+}
+
+func validateCheckRuns(checkRun *github.CheckRunOutput) error {
+	if checkRun == nil {
+		return errors.New("checkRun Output is nil")
+	}
+
+	catcher := grip.NewBasicCatcher()
+
+	catcher.NewWhen(checkRun.Title == nil, "checkRun has no title")
+	summaryErrMsg := fmt.Sprintf("the checkRun '%s' has no summary", utility.FromStringPtr(checkRun.Title))
+	catcher.NewWhen(checkRun.Summary == nil, summaryErrMsg)
+
+	for _, an := range checkRun.Annotations {
+		annotationErrorMessage := fmt.Sprintf("checkRun '%s' specifies an annotation '%s' with no ", utility.FromStringPtr(checkRun.Title), utility.FromStringPtr(an.Title))
```

Comment:
This might be okay but when title is empty, this message is empty. I don't know if there's a better way to do this though.

Reply:
Yeah, I thought about that too. I don't think there's a better way either. At least the ' ' will indicate to the user that the title is empty.